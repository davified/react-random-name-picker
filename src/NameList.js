import React from "react";

const NameList = props => (
  <div>
    {props.names.map((name, index) => {
      const isLuckyWinner = index === props.luckyWinnerIndex;
      return (
        <li id={isLuckyWinner ? "winner" : ""} key={index}>
          {name}
        </li>
      );
    })}
  </div>
);

export default NameList;
