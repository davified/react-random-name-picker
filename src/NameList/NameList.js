import React from "react";

const NameList = props => (
  <div>
    <ul>
      {props.names.map((name, i) => {
        const isWinner = props.luckyWinnerIndex === i;
        return (
          <li key={i} id={isWinner ? "winner" : ""}>
            {name}
          </li>
        );
      })}
    </ul>
  </div>
);

export default NameList;
