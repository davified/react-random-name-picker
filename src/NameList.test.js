import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import NameList from "./NameList";

test("snapshot test for NameList", () => {
  const renderer = new ShallowRenderer();
  const testNames = ["name1", "name2"];
  renderer.render(<NameList names={testNames} luckyWinnerIndex={0} />);
  const output = renderer.getRenderOutput();

  expect(output).toMatchSnapshot();
});
