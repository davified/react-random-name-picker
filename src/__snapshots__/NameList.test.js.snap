// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for NameList 1`] = `
<div>
  <ul>
    <li
      id="winner"
    >
      name1
    </li>
    <li
      id=""
    >
      name2
    </li>
  </ul>
</div>
`;
